lib/player_input.rb

```diff
@@ -0,0 +1,51 @@
+class PlayerInput
```

Comment:
This class is less intuitive to me, and that comes across with the size and complexity of the class and methods. That it needs to output and handle input seems like too much responsibility.

What about something that just wraps input. This might be two classes. One that coordinates a turn, printing the prompt and then getting/validating input, and one that wraps the input?

Reply:
Pulled out a few things in 2f3dcb7 and d733979. I like the "turn" mentioned below, so renamed this class to PlayerTurn in 875d678.